lsp/src/main/scala/org/ensime/lsp/rpc/messages.scala

```diff
@@ -0,0 +1,222 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.lsp.rpc.messages
+
+import org.ensime.lsp.rpc.JsInnerFormats.JsInnerField
+import shapeless.tag
+import shapeless.tag.@@
+import spray.json._
+
+import scala.collection.immutable.Seq
+
+object JsonRpcMessages {
```

Comment:
putting ADTs in an `object` tends to amplify bugs in the scala compiler. Best put them in a package.

Reply:
I can't write top-level type binding for `Params`